### PR TITLE
Add sprayer mission item

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

File: libraries/GCS_MAVLink/GCS_Common.cpp [hunk 1]
```diff
@@ -3710,6 +3711,20 @@ MAV_RESULT GCS_MAVLINK::handle_command_do_gripper(const mavlink_command_long_t &
...
 
+#if HAL_SPRAYER_ENABLED
+MAV_RESULT GCS_MAVLINK::handle_command_do_sprayer(const mavlink_command_long_t &packet)
+{
+    AC_Sprayer *sprayer = AP::sprayer();
+    if (sprayer == nullptr) {
+        return MAV_RESULT_FAILED;
+    }
+
+    sprayer->run(uint8_t(packet.param1));
```
Comment (rmackay9) [inline libraries/GCS_MAVLink/GCS_Common.cpp:3722]: run takes a boolean.. I'm sure this will work but I guess I would make it "is_positive(packet.param1)"

**Comment by rmackay9** [comment]
I wonder if it's good to keep putting mav-cmd into ardupilotmega.xml.  Might as well put them into common no?

**Comment by rmackay9** [comment]
This looks pretty good to me, thanks!

**Comment by auturgy** [comment]
Agree, but if we want them in common they need to go via mavlink/mavlink else qgc will never get them. We really need to keep common sync’d.

File: libraries/GCS_MAVLink/GCS_Common.cpp (hunk 1 above)
Comment (peterbarker) [reply libraries/GCS_MAVLink/GCS_Common.cpp:3722]: I might just cast this to a boolean instead.  I'd like to keep non-boolean values Reserved For Future Use, if that's OK.

**Comment by amilcarlucas** [comment]
This can now go in

File: libraries/GCS_MAVLink/GCS_Common.cpp (hunk 1 above)
Comment (peterbarker) [reply libraries/GCS_MAVLink/GCS_Common.cpp:3860]: I've adjusted this and the mission equivalent to strictly take 0 or 1 to allow for future modifications to the use of this value.

**Comment by peterbarker** [comment]
@amilcarlucas I've rebased this and addressed @rmackay9 's comments.  I think this will need to have another dev merge it as it does update the mavlink submodule.

**Comment by rmackay9** [comment]
Looks good to me, thanks!  Feel free to merge after it passes CI.

**Comment by peterbarker** [comment]
Can be merged after CI passes (careful with mavlink commit!)

**Comment by amilcarlucas** [comment]
You missed the ArduPilot/mavlink#139 PR

**Comment by CemalettinAkdogan** [comment]
Hi, I want to use command MAV_CMD_DO_SPRAY my agriculture drone in dronekit programming but I don't know how to run the command, Can you help me ?

**Comment by amilcarlucas** [comment]
Read the dronekit documentation on how to send custom mavlink messages.

**Pull request merged**